contracts/precompiles/lib/fungibleTokenPacketData.sol

```diff
@@ -0,0 +1,334 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+import "./ProtoBufRuntime.sol";
+import "./GoogleProtobufAny.sol";
+
+library FungibleTokenPacketData {
+
+
+  //struct definition
+  struct Data {
+    string denom;
+    uint64 amount;
+    bytes sender;
+    bytes receiver;
+  }
+
+  // Decoder section
+
+  /**
+   * @dev The main decoder for memory
+   * @param bs The bytes array to be decoded
+   * @return The decoded struct
+   */
+  function decode(bytes memory bs) internal pure returns (Data memory) {
```

Comment:
q: why do you need these funcs?

Reply:
Helps me decode data - I was following some code examples.

`FungibleTokenPacketData.Data memory data = FungibleTokenPacketData.decode(packet.data);`